strax/context.py

```diff
@@ -798,7 +800,9 @@ def concat_loader(*args, **kwargs):
                 if d not in targets:
                     return
             elif p.save_when == strax.SaveWhen.EXPLICIT:
-                if d not in save:
+                # If we arrive here in case of a superrun the user want to save
+                # as self.context_config['write_superruns'] is true.
+                if d not in save and not _is_superrun:
```

Comment:
Do I get it correctly that this means now we will always save it? Because if `_is_supperrun` and `p.save_when == strax.SaveWhen.EXPLICIT` we will always save it irrespective of the save argument?

Reply:
This one we can debate. So here is the example I was think of: 

The user would like to make a superrun for a SaveWhen.EXPLICIT plugin and sets `st.set_context_config({'write_superruns': True})`. If we do not put `and not _is_superrun` here and the user calls `st.make(superrun_id, target)` without `save=(target,)` the data for the subruns will be created (if the data does not exists), but the superrun will not. I thought since the user specified already `st.set_context_config({'write_superruns' he/she wishes to store the superrun and we save analysts the desperate experience in creating data for hours before they realize in the end they forgot to put the `save`-argument.

Reply:
okay, of course it shouldn't take an hour but I see. I would have just raised an error when someone would request making a superrun for an explicit plugin without the save argument but this approach also makes sense. Perhaps add a small comment for or check for the 'write_supperruns' ?

How does this work for the components in the processing chain that are not part of the request? E.g. I ask for `cut_kr` depending on `event_info_double` (savewhen.explicit), do we then also start writing `event_info_double`, I think we do right?

Reply:
> How does this work for the components in the processing chain that are not part of the request? E.g. I ask for cut_kr depending on event_info_double (savewhen.explicit), do we then also start writing event_info_double, I think we do right?

No we should not. In line 726 we process all subruns which do not exists yet by calling `make(sub_run_id, target, save_when=(target, ))` E.g. in your example we would do something like `make('0123456', cut_kr, save_when=(cut_kr, ))` . Hence `event_info` double should not be stored.

Reply:
okay thanks